sdk/eventhub/azure-eventhub/migration_guide.md

```diff
@@ -1,129 +1,179 @@
+- The `add_sender` method is no longer used to create sender clients. Instead, the `EventHubProducerClient` is used for sending events.
+- The `send` method that allowed sending single events in each call is removed in favor of the `send_batch` to encourage sending events in batches for better throughput.
+- The new `send_batch` method takes a list of `EventData` objects that is batched into a single message by the client before sending.
+- The above approach fails if the list of events increase the size limit of the message. To safely send within size limits, use the `EventDataBatch` object to which you can add `EventData` objects until the size limit is reached after which you can send it using the same `send_batch` method.
 
-So in V1:
+In v1:
 ```python
-client = EventHubClientAsync.from_connection_string(connection_str, eventhub=EVENTHUB_NAME)
-sender = client.add_async_sender(partition="0")
-try:
-    await client.run_async()
-    event_data = EventData(b"A single event")
-    await sender.send(event_data)
-finally:
-    await client.stop_async()
+client = EventHubClient(address)
+sender = client.add_sender()
+client.run()
+sender.send(EventData('Single message'))
+client.stop()
 ```
 
-In V5:
+In v5:
 ```python
-producer = EventHubProducerClient.from_connection_string(conn_str=EVENT_HUB_CONNECTION_STR, eventhub_name=EVENTHUB_NAME)
-async with producer:
-    event_data_batch = await producer.create_batch(partition_id="0")
-    event_data_batch.add(EventData(b"A single event"))
-    await producer.send_batch(event_data_batch)
+producer_client = EventHubProducerClient.from_connection_string(conn_str, eventhub_name)
+
+# Send EventDataBatch
+event_data_batch = producer.create_batch()
+event_data_batch.add(EventData('Single message'))
+producer.send_batch(event_data_batch)
+
+# Send list of EventData
+event_data_batch = [EventData('Single message')]
+producer.send_batch(event_data_batch)
 ```
 
+### Receiving events 
+
+- The `run` and `stop` methods were previously used since the single `EventHubClient` controlled the lifecycle for all senders and receivers. In v5, the `run` and `stop` methods are deprecated since the `EventHubConsumerClient` controls its own lifecycle.
+- The `add_receiver` method is no longer used to create receiver clients. Instead, the `EventHubConsumerClient` is used for receiving events.
+- The old `receive` method returned a list of `EventData`.
+- The new `receive` calls the user callback `on_event` to process single events for easier and more clear interaction with event data when dealing with multiple partitions. 
+- The new `receive_batch` calls the user callback `on_event_batch` to process batches of events for easier and more clear interaction with event data when dealing with multiple partitions.
```

Comment:
```suggestion
```suggestion
- In v1, the `receive` method returned a list of `EventData`. You would call this method repeatedly every time you want receive a set of events. In v5, the new `receive` method takes user callback to process events and any resulting errors. This way, you call the method once and it continues to process incoming events until you stop it.
- Additionally, we have a method `receive_batch` which behaves the same as `receive`, but calls the user callback with a batch of events instead of single events. 
- The same methods can be used whether you want to receive from a single partition or from all partitions.
```